app/modules/intelligence/agents_copy/custom_agents/custom_agents_service.py

```
@@ -0,0 +1,466 @@
+            "system_prompt": agent_model.system_prompt,
+            "tasks": agent_model.tasks,
+        }
+        llm = ProviderService(self.db, user_id).get_large_llm(AgentType.LANGCHAIN)
+        runtime_agent = RuntimeAgent(llm, self.db, agent_config, self.secret_manager)
+        try:
+            result = await runtime_agent.run(
+                agent_id, query, project_id, conversation_id, node_ids
+            )
+            return {"message": result["response"]}
+        except Exception as e:
+            raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)